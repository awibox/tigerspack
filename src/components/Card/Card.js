import React from 'react';
import PropTypes from 'prop-types';
import { colors } from '../../utils/colors';
import { defaultStyles } from '../../utils/defaultStyles';

const Card = (props) => {
  const {
    children,
    title,
    icon,
    theme,
    outline,
    withoutContainer,
    ...otherProps
  } = props;
  const palette = colors[theme] ? colors[theme] : colors.primary;
  const padding = props.padding || defaultStyles.indent;
  const indent = props.indent || defaultStyles.indent;
  const styles = {
    card: {
      border: outline ? 'none' : `1px solid ${palette.color}`,
      borderTop: outline && `3px solid ${palette.color}`,
      background: '#fff',
      boxShadow: '0 1px 3px rgba(0, 0, 0, 0.12)',
      boxSizing: 'border-box',
      transition: defaultStyles.animation,
      borderRadius: `${defaultStyles.borderRadius}px`,
      marginBottom: `${indent}px`,
      overflow: 'hidden',
    },
    title: {
      borderBottom: outline && '2px solid #e7eaec',
      background: !outline ? palette.color : '',
      color: !outline ? palette.text : palette.color,
      width: '100%',
      boxSizing: 'border-box',
      padding: `${padding}px`,
      fontSize: defaultStyles.mediumFontSize,
      fontWeight: '600',
      textTransform: 'uppercase',
    },
    content: {
      padding: `${padding}px`,
    },
    icon: {
      marginRight: '10px',
    },
  };
  return (
    <div css={styles.card} {...otherProps}>
      {title && <div css={styles.title}>
        {icon && <span css={styles.icon}>{icon}</span>}
        {title}
      </div>}
      {!withoutContainer ? <div css={styles.content}>{children}</div> : children}
    </div>
  );
};

Card.propTypes = {
  children: PropTypes.any.isRequired,
  icon: PropTypes.any,
  withoutContainer: PropTypes.bool,
  indent: PropTypes.number,
  padding: PropTypes.number,
  outline: PropTypes.bool,
  title: PropTypes.string,
  theme: PropTypes.string,
};

export default Card;
