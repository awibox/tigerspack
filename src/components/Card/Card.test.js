import React from 'react';
import { shallow } from 'enzyme';
import Card from './Card';

describe('Card', () => {
  it('Default Card', () => {
    const component = shallow(<Card>Test alert</Card>);
    expect(component).toMatchSnapshot();
  });
  it('Outline Card', () => {
    const component = shallow(<Card outline title={'Test title'} theme={'danger'}>Test Card</Card>);
    expect(component).toMatchSnapshot();
  });
  it('Card with icon', () => {
    const component = shallow(<Card outline title={'Test title'} icon={<span>icon</span>} theme={'danger'}>Test Card</Card>);
    expect(component).toMatchSnapshot();
  });
  it('Card without container', () => {
    const component = shallow(<Card outline withoutContainer title={'Test title'} icon={<span>icon</span>} theme={'danger'}>Test Card</Card>);
    expect(component).toMatchSnapshot();
  });
});
