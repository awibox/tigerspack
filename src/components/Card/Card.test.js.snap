// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Card with icon 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "background": "#fff",
      "border": "none",
      "borderRadius": "3px",
      "borderTop": "3px solid #D15130",
      "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "marginBottom": "15px",
      "overflow": "hidden",
      "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "background": "",
        "borderBottom": "2px solid #e7eaec",
        "boxSizing": "border-box",
        "color": "#D15130",
        "fontSize": "14px",
        "fontWeight": "600",
        "padding": "15px",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
  >
    <EmotionCssPropInternal
      __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
      css={
        Object {
          "marginRight": "10px",
        }
      }
    >
      <span>
        icon
      </span>
    </EmotionCssPropInternal>
    Test title
  </EmotionCssPropInternal>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "padding": "15px",
      }
    }
  >
    Test Card
  </EmotionCssPropInternal>
</EmotionCssPropInternal>
`;

exports[`Card Card without container 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "background": "#fff",
      "border": "none",
      "borderRadius": "3px",
      "borderTop": "3px solid #D15130",
      "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "marginBottom": "15px",
      "overflow": "hidden",
      "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "background": "",
        "borderBottom": "2px solid #e7eaec",
        "boxSizing": "border-box",
        "color": "#D15130",
        "fontSize": "14px",
        "fontWeight": "600",
        "padding": "15px",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
  >
    <EmotionCssPropInternal
      __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
      __EMOTION_TYPE_PLEASE_DO_NOT_USE__="span"
      css={
        Object {
          "marginRight": "10px",
        }
      }
    >
      <span>
        icon
      </span>
    </EmotionCssPropInternal>
    Test title
  </EmotionCssPropInternal>
  Test Card
</EmotionCssPropInternal>
`;

exports[`Card Default Card 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "background": "#fff",
      "border": "1px solid #3F88C5",
      "borderRadius": "3px",
      "borderTop": undefined,
      "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "marginBottom": "15px",
      "overflow": "hidden",
      "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "padding": "15px",
      }
    }
  >
    Test alert
  </EmotionCssPropInternal>
</EmotionCssPropInternal>
`;

exports[`Card Outline Card 1`] = `
<EmotionCssPropInternal
  __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
  __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
  css={
    Object {
      "background": "#fff",
      "border": "none",
      "borderRadius": "3px",
      "borderTop": "3px solid #D15130",
      "boxShadow": "0 1px 3px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "marginBottom": "15px",
      "overflow": "hidden",
      "transition": "all 0.3s cubic-bezier(.25, .8, .25, 1)",
    }
  }
>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "background": "",
        "borderBottom": "2px solid #e7eaec",
        "boxSizing": "border-box",
        "color": "#D15130",
        "fontSize": "14px",
        "fontWeight": "600",
        "padding": "15px",
        "textTransform": "uppercase",
        "width": "100%",
      }
    }
  >
    Test title
  </EmotionCssPropInternal>
  <EmotionCssPropInternal
    __EMOTION_LABEL_PLEASE_DO_NOT_USE__="Component"
    __EMOTION_TYPE_PLEASE_DO_NOT_USE__="div"
    css={
      Object {
        "padding": "15px",
      }
    }
  >
    Test Card
  </EmotionCssPropInternal>
</EmotionCssPropInternal>
`;
